File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        //brute-force approach
        int n = nums.size();
        sort(nums.begin(), nums.end());
        return nums[n/2]; //because majority element occurs more than n/2 times, and when the array is sorted, it will occupy the middle position.
    }
};

File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let candidate = -1;
    let votes = 0;
    for(let i = 0; i<nums.length; i++){
        if(votes == 0){
            candidate = nums[i];
            // votes = 1;
        }
        if(nums[i] == candidate){
            votes++;
        }
        else{
            votes--;
        }
    }
    return candidate;
};